include/zephyr/posix/grp.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_POSIX_GRP_H_
+#define ZEPHYR_INCLUDE_POSIX_GRP_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/posix/sys/stat.h>
+
+struct group {
```

Comment:
Could the structs and fields have proper doxygen comments

Reply:
Yes. Very good question.

I would like to do a header cleanup and add Doxygen but ideally get code PR's in first.